qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/ParameterizedRollingUpgradeTestCase.java

```diff
@@ -78,6 +80,13 @@ protected ParameterizedRollingUpgradeTestCase(@Name("upgradedNodes") int upgrade
         this.requestedUpgradedNodes = upgradedNodes;
     }
 
+    @Before
```

Comment:
The clear is an `AfterClass`. So should this be `BeforeClass`?

Reply:
It cannot be in `BeforeClass`, as it needs `testFeatureService` to be initialized - which for reasons happens in a `Before` method in the base class.
Besides, it follows the same pattern as the other initialization functions in the test, like `extractOldIndexVersion`.
My understanding is that everything is initialized in `Before` (keeping track of the upgrade "phase"), and when only when all tests from all phases are done we cleanup (that's why `AfterClass`)